pkg/config/store/store2.go

```diff
@@ -62,8 +62,31 @@ type BackEndResource struct {
 
 // Resource represents a resources with converted spec.
 type Resource struct {
-	Metadata ResourceMeta
-	Spec     proto.Message
+	Metadata ResourceMeta  `json:"metadata"`
+	Spec     proto.Message `json:"spec"`
+	Status   Status        `json:"status"`
```

Comment:
Does this need to belong to Resource?  As far as I see, the status is handled separately as `statusMap` in `pkg/runtime`, so `Resource` doesn't have to hold the status as a field?

Reply:
This is somewhat consistent with k8s, where `spec`, `metadata` and `status` are all at the same level, and status is considered a sub object of the resource.